### ISSUE TITLE
Invalid label name

### ISSUE THREAD

**Issue by caseydavenport** [coreos/prometheus-operator#78]
I'm seeing what I assume to be an incompatibility between k8s label syntax and prometheus label syntax.

```
time="2016-12-29T22:24:07Z" level=info msg="Starting prometheus (version=1.3.0, branch=master, revision=18254a172b1e981ed593442b2259bd63617d6aca)" source="main.go:75"
time="2016-12-29T22:24:07Z" level=info msg="Build context (go=go1.7.3, user=root@d363f050a0e0, date=20161101-17:06:27)" source="main.go:76"
time="2016-12-29T22:24:07Z" level=info msg="Loading configuration file /etc/prometheus/config/prometheus.yaml" source="main.go:247"
time="2016-12-29T22:24:07Z" level=error msg="Error loading config: couldn't load configuration (-config.file=/etc/prometheus/config/prometheus.yaml): \"__meta_kubernetes_service_label_k8s-app
\" is not a valid label name" source="main.go:149"
```

Seems reinforced by this: https://github.com/prometheus/prometheus/issues/1178

Not sure if there is scope for trying to work around this through escaping the `-` character (and others) somehow in the prometheus operator, or if this is just "live with" for now.

**Comment by brancz** [comment]
Which version of the operator are you using? This should be fixed in v0.1.1 and v0.2.1 respectively.

**Comment by caseydavenport** [comment]
Looks like I was way out of date, still running v0.0.1 :)

Thanks!
